src/Chainweb/BlockHash.hs

```diff
@@ -236,21 +236,21 @@ decodeBlockHashRecordChecked ps = do
     hashes <- mapM decodeBlockHashWithChainIdChecked (Expected <$!> getExpected ps)
     return $! BlockHashRecord $! HM.fromList hashes
 
-blockHashRecordToSequence :: BlockHashRecord -> S.Seq BlockHash
-blockHashRecordToSequence = S.fromList . fmap snd . sort . HM.toList . _getBlockHashRecord
+blockHashRecordToVector :: BlockHashRecord -> V.Vector BlockHash
+blockHashRecordToVector = V.fromList . fmap snd . sort . HM.toList . _getBlockHashRecord
```

Comment:
Here it's quite a lot cheaper to sort vector-side with `vector-algorithms` (especially for small lists), and you can probably use `unsafeThaw`/`unsafeFreeze` to sort in place without copying the input. BTW part of my issue here is that upstream a lot of the things that are in hashmaps (like cuts, the mapping from chain id -> `ChainResources`), etc are `HashMap ChainId Foo` when they should be a newtype over sorted `Vector (ChainId, Foo)`.

Reply:
Agreed, that this is the right thing to do. Since we currently don't have vector-algorithms in the list of dependencies, and this code here is already tested and doesn't seem to be a major performance concern at this point, I propose that we keep those ideas in mind for follow up work.